packages/main/src/plugin/index.ts

```diff
@@ -306,6 +309,86 @@ export class PluginSystem {
       'feedback',
       undefined,
     );
+
+    if (import.meta.env.PROD) {
+      // Get the current version of our application
+      const currentVersion = `v${app.getVersion()}`;
+
+      // Add update entry to the status bar
+      // this will show the current version, toolbar for the version number as well.
+      statusBarRegistry.setEntry(
+        'version',
+        false,
+        0,
+        currentVersion,
+        `Using version ${currentVersion}`,
+        'fa-light fa-circle-info',
+        true,
+        'version',
+        undefined,
+      );
+
+      // Only import on production builds
+      import('electron-updater').then(({ autoUpdater }) => {
+        // Setting the feed URL is now required as we are now using
+        // autoUpdater.on('update-available', () => {
+        // in other parts of the code
+        autoUpdater.setFeedURL({
+          provider: UPDATER_PROVIDER,
+          owner: UPDATER_OWNER,
+          repo: UPDATER_REPO,
+        });
+
+        // autoUpdater.checkForUpdatesAndNotify() is called from main/src/index.ts
+        autoUpdater.on('update-available', () => {
+          // If there is an update available, remove the "version" entry from the status bar
+          // and add the "update" entry
+          statusBarRegistry.removeEntry('version');
+
+          // Will add the new entry.
+          // If there is already an entry available, it'll update the icon to the new one (if it's different)
+          statusBarRegistry.setEntry(
+            'update',
+            false,
+            0,
+            currentVersion,
+            'Update available',
+            UPDATER_UPDATE_AVAILABLE_ICON,
+            true,
+            'update',
+            undefined,
+          );
+        });
```

Comment:
I'm wondering why we don't have a single entry that do all the stuff ?
why add a new item and remove the previous one. we can just handle only one item in the status bar and depending on the context, update the icon/text, etc ?

Reply:
> I'm wondering why we don't have a single entry that do all the stuff ? why add a new item and remove the previous one. we can just handle only one item in the status bar and depending on the context, update the icon/text, etc ?

Instead of removing / adding, would it make more sense if we just set enabled / disabled? We could do that instead. Setup both items at the beginning and then enable / disable in the logic.